lib/dao.js

```diff
@@ -1663,7 +1663,7 @@ DataAccessObject._coerce = function(where, options) {
   }
 
   var props = self.definition.properties;
-  for (var p in where) {
+  eachprop: for (var p in where) {
```

Comment:
I personally find jump labels an anti-pattern. Could you please refactor this code by extracting a nested function for this outer loop body?

```js
for (var p in where) {
  this._coerceProperty(p/*, ...*/);
}

// in _coerceProperty
// ...
while (parts.length) {
  var part = parts.shift();
  if (typeof obj[part].type !== 'function' ||
      !obj[part].type.definition ||
      !obj[part].type.definition.properties) {
    return;
  }
  obj = obj[part].type.definition.properties;
}
// ...
```